systemtests/bin/setup_download_tests.py

```diff
@@ -0,0 +1,306 @@
+        if sym_filename.endswith(".0"):
+            # Skip these because there aren't SYM files for them.
+            continue
+
+        is_try = False
+
+        if sym_filename.startswith("try/"):
+            sym_filename = sym_filename[4:]
+            is_try = True
+
+        if sym_filename.startswith("v1/"):
+            sym_filename = sym_filename[3:]
+
+        platform = check_platform(sym_filename)
+        build_list_of_sym_filenames(
+            sym_filename,
+            platform,
+            is_try,
+            sym_file_type_to_filename,
+        )
+
+        if len(list(sym_file_type_to_filename.values())) >= end_condition_num_files:
+            break
+
+
+def build_list_of_sym_filenames(
+    sym_filename, platform, is_try, sym_file_type_to_filename
+):
+    """
+    Builds up a sym_file_type_to_filename map of the form:
+
+    {
+        try: {filename1},
+        linux: {filename2},
+        mac: {filename3},
+        windows: {filename4},
+        unknown: {filename5}
+    }
+
+    Where each filename occurs at most once as a value in the map, and the
+    only try file is filename1.
+    """
+    if platform == "unknown" and "unknown" in sym_file_type_to_filename:
+        return
+
+    # The same file could be both a try file and a platform file,
+    # so check for both (try first).
+    if is_try:
+        if "try" not in sym_file_type_to_filename:
+            click.echo(
+                f"Adding {sym_filename} to the list as bucket: try, platform: {platform} ...",
+            )
+            sym_file_type_to_filename["try"] = sym_filename
+            return
+        # If we ensure any of our platform file types (mac, linux, windows) are from the
+        # regular bucket only, we can guarantee we get the download URL right based on
+        # the correct bucket downstream.
+        return
+
+    if platform in sym_file_type_to_filename:
+        return
+
+    click.echo(
+        f"Adding {sym_filename} to the list as bucket: regular, platform: {platform} ..."
+    )
+    sym_file_type_to_filename[platform] = sym_filename
+
+
+def write_list_of_sym_filenames_to_csv(sym_file_type_to_filename, outputdir):
+    file_types = list(sym_file_type_to_filename.keys())
+    file_types.sort()
+    REQUIRED_FILE_TYPES.sort()
+    if file_types != REQUIRED_FILE_TYPES:
+        raise KeyError(
+            f"Missing file types for download test ... Needed: {REQUIRED_FILE_TYPES} Have: {file_types}."
+        )
+
+    click.echo(f"Writing SYM filenames to CSV in {outputdir} ...")
+    csv_rows = []
+    date = datetime.datetime.now().strftime("%Y-%m-%d")
+    csv_rows.append([f"# File built: {date}"])
+    csv_rows.append(["# Format: symfile", "expected status code", "bucket", "platform"])
+    csv_rows.append(["@ECHO NOTE: None of these should take over 1000ms to download."])
+    csv_rows.append(["@ECHO"])
+    csv_rows.append(["# These are recent uploads", " so they're 200s."])
```

Comment:
Because there's a comma in the comment itself, which is a delimiter in a CSV.

The default behavior is for the CSV writer to quote any value with a delimiter character, so if I have something like:
```python
    csv_rows.append(["# These are recent uploads, so they're 200s."])
```
I get something like this output to the CSV (containing double quotes, which I don't want):
```csv
"# These are recent uploads, so they're 200s."
```
This was one way to work around removing those quotes.

Another option I tried was to pass in some config options into `csv.writer()` for `escapechar` and `quoting`, but I got some unexpected (and undesired) results with that:

```python
with open("test1.csv", "w") as f:
    csv.writer(f, escapechar="/", quoting=csv.QUOTE_NONE).writerows([["test/, test"]])
```

results in:
```
test///, test
```

Another option I haven't tried is to use `file_pointer.write()` and forgo using the `csv` library entirely, since CSVs don't have comments, so we're kind of fighting against its implementation this way. Would you prefer I do that instead?

Reply:
Ahh... got it. So the problem is that this csv file has non-csv lines in it. The non-csv lines are used by the thing using the csv file to produce output indicating what should be happening next. It makes it easier to read the systemtest output and know whether you're seeing good things or bad things.

I think there are a few things we could do.

1. Not use csv. Maybe there's some other kind of file format where we can mix echo lines and test case lines.
2. Instead of doing `write.writerows`, iterate over the `csv_rows` array and use `write.writerow` for csv lines and `f.write` for non-csv lines. You can differentiate between the two kinds of lines by looking at the first character.

For example, this works:

```python
import csv

csv_rows = [
    ["# windows files"],
    ["filename.sym", "200", "publicbucket", "windows"],
    ["filename2.sym", "200", "publicbucket", "windows"],
    ["# osx files"],
    ["filename3.sym", "200", "publicbucket", "osx"],
]

with open("output.csv", "w", newline="") as fp:
    writer = csv.writer(fp, lineterminator="\n")
    for row in csv_rows:
        if row[0].startswith("#"):
            fp.write(f"{row[0]}\n")
        else:
            writer.writerow(row)
```

Also, note we need to pass in `newline=""` to `open` per the csv docs (which are poorly written) and we need to pass in `lineterminator="\n"` to `csv.writer` otherwise we get Windows line endings.

Reply:
Thanks! I'll go with your `csv` based suggestions.